src/test/suite/lib/run-ggshield.test.ts

```diff
@@ -29,14 +27,49 @@ suite("runGGShieldCommand", () => {
     );
 
     // Assert that spawnSync was called
-    assert(spawnSyncSpy.called, "spawnSync should be called once");
+    assert(spawnSyncMock.called, "spawnSync should be called once");
 
     // Check the arguments passed to spawnSync
-    const spawnSyncArgs = spawnSyncSpy.lastCall.args;
+    const spawnSyncArgs = spawnSyncMock.lastCall.args;
     const options = spawnSyncArgs[2];
     assert.strictEqual(options.env.TEST_GLOBAL_VAR, "GlobalValue");
 
-    simple.restore();
     delete process.env.TEST_GLOBAL_VAR;
   });
+
+  const testCasesAllowSelfSigned = [
+    {
+      allowSelfSigned: true,
+      description:
+        "GGshield is called with flag --allow-self-signed when allowSelfSigned is true",
+    },
+    {
+      allowSelfSigned: false,
+      description:
+        "GGshield is not called with flag --allow-self-signed when allowSelfSigned is false",
+    },
+  ];
+
+  testCasesAllowSelfSigned.forEach(({ allowSelfSigned, description }) => {
+    test(description, async () => {
+      process.env.TEST_GLOBAL_VAR = "GlobalValue";
+
+      runGGShield.runGGShieldCommand(
+        {
+          ggshieldPath: "path/to/ggshield",
+          apiUrl: "",
+          apiKey: "",
+          allowSelfSigned: allowSelfSigned,
+        } as GGShieldConfiguration,
+        []
+      );
+
+      assert(spawnSyncMock.called, "spawnSync should be called once");
+
+      const spawnSyncArgs = spawnSyncMock.lastCall.args;
+      const args = spawnSyncArgs[1];
+
+      assert.strictEqual(args.includes("--allow-self-signed"), allowSelfSigned);
```

Comment:
This test works great.
However, I think that we should also test that the allow self signed option is passed at the correct place, since it could make the command fail.